website/pages/docs/commands/volume/register.mdx

```diff
@@ -0,0 +1,90 @@
+---
+layout: docs
+page_title: 'Commands: volume register'
+sidebar_title: register
+description: |
+  Register volumes with CSI plugins.
+---
+
+# Command: volume register
+
+The `volume register` command registers external storage volumes with
+Nomad's [Container Storage Interface (CSI)][csi] support. The volume
+must exist on the remote storage provider before it can be registered
+and used by a task.
+
+## Usage
+
+```plaintext
+nomad volume register [options] [file]
+```
+
+The `volume register` command requires a single argument, specifying
+the path to a file containing a valid [volume
+specification][volume_specification]. This file will be read and the
+job will be submitted to Nomad for scheduling. If the supplied path is
+"-", the job file is read from STDIN. Otherwise it is read from the
+file at the supplied path.
+
+## General Options
+
+@include 'general_options.mdx'
+
+## Volume Specification
+
+The file may be provided as either HCL or JSON. An example HCL configuration:
+
+```hcl
+id = "ebs_prod_db1"
+name = "database"
+type = "csi"
+external_id = "vol-23452345"
+plugin_id = "ebs-prod"
+access_mode = "single-node-writer"
+attachment_mode = "file-system"
+mount_options {
+   fs_type = "ext4"
+   mount_flags = ["ro"]
+}
+```
+
+## Volume Specification Parameters
+
+- `id` `(string: <required>)` - The unique ID of the volume. This will
+  be how [`volume`][csi_volume] stanzas in a jobspec refer to the volume.
+
+- `name` `(string: <required>)` - The display name of the volume.
+
+- `type` `(string: <required>)` - The type of volume. Currently only
+  `"csi"` is supported.
+
+- `external_id` `(string: <required>)` - The ID of the physical volume
```

Comment:
It might be better to leave this in the docs and make it required, but as it is `external_id` is used if specified but you can just use the `id` for the volume and leave it out.

Reply:
TIL. 😀 Let's keep it for now and then mark it as not required as we refine the UX of this whole thing.